flash/flashtext_test.go

```diff
@@ -132,6 +132,21 @@ func TestCaseSensitive(t *testing.T) {
 	t.Logf("allkeys: %v", allKeys)
 }
 
+func TestFalseCaseInsensitiveSearchUnicodeDoesNotPanic(t *testing.T) {
+	trie := NewFlashKeywords(false)
+	keys := []string{"İUseUnicode", "İUseUnicodeToBreakThings"}
+	for _, k := range keys {
+		trie.Add(k)
+	}
+	text := "İUseUnicode"
+	res := trie.Search(text)
+	assert.Equal(t, len(res), len(keys))
+	for i := 0; i < len(keys); i++ {
+		assert.Equal(t, res[i].Key, strings.ToLower(keys[i]))
+	}
```

Comment:
This test case will fail because the `res` slice contains only 1 element `Search Result: [{iuseunicode false  0 10}]` and `len(keys) = 2`, but apart from this small change the test case passes when the length of text is taken after lowercasing the text.

Reply:
I may be missing something - is there something that needs to be changed here to get this merged?

Reply:
This `assert.Equal(t, len(res), len(keys))` is not true, because the `text := "İUseUnicode"` that you are searching has only 1 key from the dictionary of keys that you added in the trie. When I run the test in my local it fails.

Reply:
Thank you - I've updated the tests to fix this.